pyccel/parser/semantic.py

```diff
@@ -2887,14 +2990,35 @@ def _visit_FunctionCall(self, expr):
                 return getattr(self, annotation_method)(expr)
 
         args = self._handle_function_args(expr.args)
+
         # Correct keyword names if scope is available
         # The scope is only available if the function body has been parsed
         # (i.e. not for headers or builtin functions)
-        if isinstance(func, FunctionDef) and func.scope:
+        if (isinstance(func, FunctionDef) and func.scope) or (isinstance(func, Interface) and func.is_annotated):
+            scope = func.scope if isinstance(func, FunctionDef) else func.functions[0].scope
             args = [a if a.keyword is None else \
-                    FunctionCallArgument(a.value, func.scope.get_expected_name(a.keyword)) \
+                    FunctionCallArgument(a.value, scope.get_expected_name(a.keyword)) \
                     for a in args]
-
+            func_args = func.arguments if isinstance(func,FunctionDef) else func.functions[0].arguments
+            if not func.is_annotated:
+                # Correct func_args keyword names
+                func_args = [FunctionDefArgument(AnnotatedPyccelSymbol(scope.get_expected_name(a.var.name), a.annotation),
+                            annotation=a.annotation, value=a.value, kwonly=a.is_kwonly, bound_argument=a.bound_argument)
+                            for a in func_args]
```

Comment:
here we assume that the only case to get an Interface is if it is annotated, the other case can be both, where it is not annotated if it is called for the first time and annotated if called a second time, and so on

Reply:
If it is an assumption then why do you need to test for it? I'm still not sure why you can ignore the case of an unannotated Interface here

Reply:
Ah! Of course. So then why do you need to check if it is annotated rather than just doing:
```python
if (isinstance(func, FunctionDef) and func.scope) or isinstance(func, Interface):
```

Reply:
sure